launcher/main.go

```diff
@@ -286,6 +317,51 @@ func (dbus *Dbus) Close() {
 	dbus.Cmd.Wait()
 }
 
+type WaylandProxy struct {
+	Cmd        *exec.Cmd
+	SocketPath string
+}
+
+func StartWaylandProxy(conf Config) (waylandProxy WaylandProxy) {
+	failed := true
+
+	cmd := exec.Command(conf.WaylandProxyExe, conf.WaylandProxyArgs...)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+
+	waylandProxy.Cmd = cmd
+	waylandProxy.SocketPath = conf.WaylandProxySocketPath
+
+	if err := cmd.Start(); err != nil {
+		panic(err)
+	}
+	defer func() {
+		if failed {
+			waylandProxy.Close()
+		}
+	}()
+
+	failed = false
+	return
+}
+
+func (waylandProxy *WaylandProxy) WaitUntilStartup() {
```

Comment:
Can we do this without polling?

Reply:
The problem is that wayland-proxy-virtwl does not provide any feedback about it's startup other than that the socket appears.
It only logs about the startup when verbose mode is enabled, which also logs all wayland requests and responses. Therefore, I do not really want to go into that direction.

One other option would be to use inotify. However, I think it is not easily usable via the golang standard library (maybe it would be possible to manually craft the syscalls - but I also do not really want to do that).
Therefore, we would need to add an external dependency, for example fsnotify (https://github.com/fsnotify/fsnotify).
Since it would be the first (non-Google) external dependency, I tried to come up with this simple pooling solution.

Reply:
fsnotify looks like a small enough library, it's fine if we include it.